v2/pkg/protocols/others/websocket/websocket.go

```diff
@@ -0,0 +1,365 @@
+		if dataErr != nil {
+			r.options.Output.Request(r.options.TemplateID, input, "websocket", dataErr)
+			r.options.Progress.IncrementFailedRequestsBy(1)
+			return errors.Wrap(dataErr, "could not evaluate template expressions")
+		}
+		reqBuilder.WriteString(string(finalData))
+
+		err = wsutil.WriteClientMessage(conn, ws.OpText, finalData)
+		if err != nil {
+			r.options.Output.Request(r.options.TemplateID, input, "websocket", err)
+			r.options.Progress.IncrementFailedRequestsBy(1)
+			return errors.Wrap(err, "could not write request to server")
+		}
+
+		msg, _, err := wsutil.ReadServerData(conn)
+		if err != nil {
+			r.options.Output.Request(r.options.TemplateID, input, "websocket", err)
+			r.options.Progress.IncrementFailedRequestsBy(1)
+			return errors.Wrap(err, "could not write request to server")
+		}
+
+		responseBuilder.Write(msg)
+		if req.Name != "" {
+			bufferStr := string(msg)
+			if req.Name != "" {
+				inputEvents[req.Name] = bufferStr
+			}
+
+			// Run any internal extractors for the request here and add found values to map.
+			if r.CompiledOperators != nil {
+				values := r.CompiledOperators.ExecuteInternalExtractors(map[string]interface{}{req.Name: bufferStr}, protocols.MakeDefaultExtractFunc)
+				for k, v := range values {
+					dynamicValues[k] = v
+				}
+			}
+		}
+	}
+	r.options.Progress.IncrementRequests()
+
+	if r.options.Options.Debug || r.options.Options.DebugRequests {
+		requestOutput := reqBuilder.String()
+		gologger.Info().Str("address", input).Msgf("[%s] Dumped Websocket request for %s", r.options.TemplateID, input)
+		gologger.Print().Msgf("%s", requestOutput)
+	}
+
+	r.options.Output.Request(r.options.TemplateID, input, "websocket", err)
+	gologger.Verbose().Msgf("Sent Websocket request to %s", input)
+
+	data := make(map[string]interface{})
+	for k, v := range previous {
+		data[k] = v
+	}
+	for k, v := range dynamicValues {
+		data[k] = v
+	}
+	for k, v := range inputEvents {
+		data[k] = v
+	}
+	data["success"] = "true"
+	data["request"] = reqBuilder.String()
+	data["response"] = responseBuilder.String()
+	data["host"] = input
+	data["ip"] = r.dialer.GetDialedIP(hostname)
+
+	event := eventcreator.CreateEventWithAdditionalOptions(r, data, r.options.Options.Debug || r.options.Options.DebugResponse, func(internalWrappedEvent *output.InternalWrappedEvent) {
+		internalWrappedEvent.OperatorsResult.PayloadValues = dynamicValues
+	})
+	if r.options.Options.Debug || r.options.Options.DebugResponse {
+		responseOutput := responseBuilder.String()
+		gologger.Debug().Msgf("[%s] Dumped Websocket response for %s", r.options.TemplateID, input)
+		gologger.Print().Msgf("%s", responsehighlighter.Highlight(event.OperatorsResult, responseOutput, r.options.Options.NoColor))
+	}
+
+	callback(event)
+	return nil
+}
+
+// getAddress returns the address of the host to make request to
+func getAddress(toTest string) (string, error) {
+	if !strings.HasPrefix(toTest, "ws://") && !strings.HasPrefix(toTest, "wss://") {
```

Comment:
nothing guarantees that the input is always lower-cased. You could first try to parse the input and then check the whether the `url.Scheme` is equal to `ws` or `wss`, because that's already lower-cased